lib/server/worker.js

```diff
@@ -58,7 +61,12 @@ exports.create = function() {
     path: '/a/{id}',
     config: {
       handler: function delete_(req, reply) {
-        img.delete(req.params.id).done(reply, reply);
+        P.all(Object.keys(SIZES).map(function(name) {
+          if (name === 'default') {
+            return req.params.id;
+          }
+          return req.params.id + '_' + name;
+        })).map(img.delete).done(reply, reply);
```

Comment:
Given the possibility for failures in both upload and in cleanup, I think we should translate a `404 Not Found` failure on delete into "sure, yeah, I deleted that, no errors here" in this API.